### ISSUE TITLE
Match node on multiple labels performance regression in Neo4j 5

### ISSUE THREAD

**Issue by ikwattro** [neo4j/neo4j#13210]
In Neo4j 5, the following query 

```
MATCH (n:Person:Active)
RETURN count(n)
```
where I have 25 million `Person` nodes but only 10 have the `Active` label is doing a scan on 25 million Person nodes while in Neo4j 4.4 its taking the label with the lowest amount of nodes.

**Neo4j Version:** 5.8.0-enterprise
**Operating System:** Ubuntu
**API:** Irrelevant

### Steps to reproduce

Create a graph with 25M Person nodes, add the Active label to 10 of them

### Expected behavior

Same behaviour as in Neo4j 4.4

<img width="393" alt="Screenshot 2023-06-11 at 12 28 22" src="https://github.com/neo4j/neo4j/assets/1222009/0e3eed2c-9e7d-40ed-b952-860478829a5c">

### Actual behavior

IntersectionNodesByLabelScan on 25M Person nodes

<img width="305" alt="Screenshot 2023-06-11 at 12 23 35" src="https://github.com/neo4j/neo4j/assets/1222009/e8f068c0-5cf1-4819-a140-8f8953fa4712">

**Comment by jexp** [comment]
Yep seems to not to pick the label index with the highest selectivity.

**Comment by nadja-muller** [comment]
Thank you, we will investigate this.

**Comment by blouerat** [comment]
Hi @ikwattro,

In your second screenshot, the `IntersectionNodeByLabelsScan` is returning _35,439_ rows, it really should return 10 rows based on the description of the scenario.
Was it taken on a different data set for illustration purposes or did it genuinely return 35k rows instead of 10? If so, what was the final count?
This is key because in one case we have a cost estimation issue, whereas in the other one `IntersectionNodeByLabelsScan` is misbehaving completely.

**Comment by ikwattro** [comment]
@blouerat Sorry, was another db test, but result is same

![Screenshot 2023-06-13 at 11 58 42](https://github.com/neo4j/neo4j/assets/1222009/519e6598-ccc4-4780-a19d-b821cfc4854e)

**Comment by blouerat** [comment]
Perfect, thanks for confirming, it is indeed a cost estimation issue, we are looking into it.

**Comment by blouerat** [comment]
@ikwattro  While we're working on working on tuning the cost estimation for `IntersectionNodeByLabelsScan`, you can tell the planner not to use it using an internal configuration option:
```
internal.cypher.planning_intersection_scans_enabled=false
```
Note that this is an undocumented option, and so it may be removed or changed in a future version without warning. 🤐

**Comment by inmost-light** [comment]
@ikwattro
Thank you for reporting, a fix has been implemented and should be available in 5.10.
